client/init.js

```diff
@@ -87,17 +88,45 @@ addEventListener('load', function() {
 
 	function startWatchingPosition() {
 		navigator.geolocation.watchPosition(function(position) {
-			try {
-				comms.send({
-					event: 'friend-updated',
-					position: position
-				})
-			} catch (e) {
-				console.error(e)
-			}
+			sendMessage({
+				event: 'friend-updated',
+				position: position
+			})
 		})
 	}
 
+	function setupHiddenEvent(){
+		var visibility = new Visibility()
+		function handleVisibilityChange(){
+			var hiding = document[visibility.hidden]
+			if (hiding){
+				timeoutId = window.setTimeout(function(){
+					sendMessage({
+						event: 'friend-updated',
+						stale: true,
+						hiding: true
+					})
+				}, timeout)
+			}
+			else
+				window.clearTimeout(timeoutId)
+			sendMessage({
+				event: 'friend-updated',
```

Comment:
I like how redux handles event propagation, but it's heavily reliant on module imports on both client and server.